src/preFilter.ts

```diff
@@ -65,12 +65,12 @@ export function filterByContent(searchee: Searchee): boolean {
 export function releaseInBlockList(
 	searchee: Searchee,
 	blockList: string[]
-): boolean {
+): string | boolean {
 	return blockList.some((blockedStr) => {
-		return (
-			searchee.name.includes(blockedStr) ||
+		return searchee.name.includes(blockedStr) ||
 			blockedStr === searchee.infoHash
-		);
+			? blockedStr
+			: false;
 	});
 }
```

Comment:
maybe string | undefined or string | null would be a more apt return type for this.

Reply:
`some` returns a boolean, so I'd have to change it to something like this?

```ts
export function releaseInBlockList(
	searchee: Searchee,
	blockList: string[]
): string | undefined {
	let foundString: string | undefined;

	blockList.some((blockedStr) => {
		if (
			searchee.name.includes(blockedStr) ||
			blockedStr === searchee.infoHash
		) {
			foundString = blockedStr;
			return true;
		}
		return false;
	});

	return foundString;
}
```

Reply:
ah I think this has a bug then. `some` doesn't return the first value you return it that's truthy, it just returns `true`
<img width="293" alt="image" src="https://github.com/cross-seed/cross-seed/assets/9119498/39d9e0fd-c977-43fb-89f7-f0a2714fad7a">

I think Array.prototype.find might be what you need

Reply:
```ts
export function releaseInBlockList(
	searchee: Searchee,
	blockList: string[]
): string | undefined {
	return blockList.find((blockedStr) => {
		return (
			searchee.name.includes(blockedStr) ||
			blockedStr === searchee.infoHash
		);
	});
}
```

Reply:
I guess I just assumed it was truthy because it was a string, not because it was actually `true`...my bad